### PR TITLE
Support macOS vibrancy

### PR DESCRIPTION
**Pull request by youngsing**
Support macOS vibrancy.
Maybe need export a function to let user specify [`NSVisualEffectMaterial`](https://developer.apple.com/documentation/appkit/nsvisualeffectview/material)? 
But now the effect has met my needs.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -72,6 +76,9 @@ impl Vibrancy for TauriWindow {
     fn apply_blur(&self) {
         #[cfg(target_os = "windows")]
         windows::apply_blur(HWND(self.hwnd().unwrap() as _));
+
+        #[cfg(target_os = "macos")]
+        macos::apply_blur(self.ns_window().unwrap() as _);
```
Comment (amrbashir) [inline src/lib.rs:81]: this should be in a separate function called `apply_vibrancy()`

File: src/lib.rs [hunk 2]
```diff
@@ -84,5 +91,8 @@ impl Vibrancy for TaoWindow {
     fn apply_blur(&self) {
         #[cfg(target_os = "windows")]
         windows::apply_blur(HWND(self.hwnd() as _));
+
+        #[cfg(target_os = "macos")]
+        macos::apply_blur(self.ns_window() as _);
```
Comment (amrbashir) [inline src/lib.rs:96]: Same as above

File: src/platform/macos.rs [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+        NSAutoresizingMaskOptions, NSView, NSViewHeightSizable, NSViewWidthSizable, NSWindow,
+        NSWindowOrderingMode,
+    },
+    base::{id, nil, BOOL},
+    foundation::{NSAutoreleasePool, NSPoint, NSRect, NSSize},
+};
+use objc::{class, msg_send, sel, sel_impl};
+
+#[allow(deprecated)]
+pub fn apply_blur(window: id) {
```
Comment (amrbashir) [inline src/platform/macos.rs:18]:
```suggestion
pub fn apply_vibrancy(window: id, appearance: MacVibrancy) {
```
where `MacVibrancy` is just an export of `NSVisualEffectMaterial` enum under a new name.

File: src/platform/macos.rs [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    unsafe fn setBlendingMode_(self, mode: NSVisualEffectBlendingMode) {
+        msg_send![self, setBlendingMode: mode]
+    }
+}
+
+#[allow(deprecated)]
+pub fn apply_blur_with_material(window: id, material: NSVisualEffectMaterial) {
+    unsafe {
+        if NSAppKitVersionNumber < NSAppKitVersionNumber10_10 {
+            // panic!("Sorry, 'NSVisualEffectView' is only available on macOS 10.10 or newer");
```
Comment (amrbashir) [inline src/platform/macos.rs:175]: maybe replace this with a debug message instead

**Review by amrbashir — CHANGES_REQUESTED** [review]
I can't thank you enough for this. Just some nitpicks and we are good to go.

File: README.md [hunk 5]
```diff
@@ -31,10 +31,10 @@
 > Please read the methods documentation, it has valuable info
 - `apply_blur()` - **`Windows Only`**
 - `apply_acrylic()` - **`Windows Only`**: works only on Windows 10 v1809 and above, it also has bad performance when resizing/dragging the window
+- `apply_vibrancy()` - **`macOS Only`**, thanks to @youngsing
```
Comment (amrbashir) [inline README.md:34]:
```suggestion
- `apply_vibrancy()` - **`macOS Only`**, thanks to [@youngsing](https://github.com/youngsing)
```

**Comment by youngsing** [comment]
I updated the code, maybe you can review it again when you have time. 
Thanks for your guidance.

**Comment by amrbashir** [comment]
@youngsing I pushed some changes to this PR to only allow the relevant methods on supported platforms, and updated README and docs, can you test if it still works perfectly on macOS before I merge?

**Comment by youngsing** [comment]
@amrbashir Yeah, it still works on Tao/Tauri windows. See the screenshot at the bottom.
But I got warnings like below:
```
warning: private type `NSVisualEffectState` in public interface (error E0446)
   --> /Users/youngsing/Code/myspace/github.com/tauri-plugin-vibrancy/src/platform/macos.rs:139:5
    |
139 |     unsafe fn setState_(self, state: NSVisualEffectState);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
140 |     unsafe fn setBlendingMode_(self, mode: NSVisualEffectBlendingMode);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(private_in_public)]` on by default
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #34537 <https://github.com/rust-lang/rust/issues/34537>
```

So we can remove `pub` from `NSVisualEffectView` to fix it? It seems that there is no need to export it.


Screen Shot 2021-12-24 at 10 06 27
Screen Shot 2021-12-24 at 10 03 05
.

**Comment by amrbashir** [comment]
Thanks, merging now.

**Pull request merged**